mailpoet/tests/integration/AutomaticEmails/WooCommerce/Events/PurchasedProductTest.php

```diff
@@ -290,83 +202,84 @@ public function _runTestItSchedulesEmailForState($state) {
-    $subscriber = Subscriber::createOrUpdate(Fixtures::get('subscriber_template'));
-    $subscriber->email = $customerEmail;
-    $subscriber->isWoocommerceUser = 1;
-    $subscriber->status = Subscriber::STATUS_SUBSCRIBED;
-    $subscriber->save();
 
-    $subscriberSegment = SubscriberSegment::create();
-    $subscriberSegment->hydrate([
-      'subscriber_id' => $subscriber->id,
-      'segment_id' => Segment::getWooCommerceSegment()->id,
-      'status' => Subscriber::STATUS_SUBSCRIBED,
-    ]);
-    $subscriberSegment->save();
+    return $newsletter;
+  }
 
+  private function createOrderEvent(Subscriber $customer, array $purchasedProductIds) {
     $orderDetails = Stub::make(
       new OrderDetails(),
       [
-        'get_billing_email' => 'test@example.com',
-        'get_items' => function() use ($incorrectProductId, $productId) {
-          return [
-            Stub::make(
+        'get_billing_email' => $customer->get('email'),
+        'get_items' => function() use ($purchasedProductIds) {
+          return array_map(function($id) {
+            return Stub::make(
               \WC_Order_Item_Product::class,
               [
-                'get_product_id' => $incorrectProductId,
+                'get_product_id' => $id,
               ]
-            ),
-            Stub::make(
-              \WC_Order_Item_Product::class,
-              [
-                'get_product_id' => $productId,
-              ]
-            ),
-          ];
+            );
+          }, $purchasedProductIds);
         },
       ]
     );
     $orderDetails->total = 'order_total';
-    $orderId = 12;
     $helper = Stub::make(WCHelper::class, [
       'wcGetOrder' => $orderDetails,
     ]);
 
-    $event = new PurchasedProduct($helper);
+    return new PurchasedProduct($helper);
+  }
 
-    // ensure there are no existing scheduled tasks
-    $scheduledTask = Sending::getByNewsletterId($newsletter->id);
-    expect($scheduledTask)->false();
+  private function triggerEmailForState(
+    string $wooOrderState,
+    Newsletter $newsletter,
+    PurchasedProduct $purchasedProductEvent
+  ) {
+    // Order ID isn't important because we're mocking wcGetOrder in the purchase event to always return the same thing
+    $orderId = 12;
 
-    // when 'woocommerce_order_status_$state' hook is triggered, an email should be scheduled
-    WPFunctions::get()->doAction('woocommerce_order_status_' . $state, $orderId);
-    $event->scheduleEmailWhenProductIsPurchased($orderId);
-    $scheduledTask = Sending::getByNewsletterId($newsletter->id);
-    expect($scheduledTask)->notEmpty();
-    $queue = $scheduledTask->queue();
-    expect($queue->getMeta())->equals(['orderedProducts' => [$incorrectProductId, $productId]]);
-    return $orderId;
+    // when 'woocommerce_order_status_$state' hook is triggered an email is scheduled when appropriate
+    WPFunctions::get()->doAction('woocommerce_order_status_' . $wooOrderState, $orderId);
```

Comment:
This is from the original version of the tests but I'm not understand why it was there to begin with. The original tests still passed with this line commented out.